internal/cli/pipeline_run.go

```diff
@@ -104,26 +104,47 @@ func (c *PipelineRunCommand) Run(args []string) int {
 		}
 
 		step.Update("Pipeline %q has started running. Attempting to read job stream sequentially in order", pipelineIdent)
+
+		steps := len(resp.JobMap)
+		step.Update("%d steps detected, run sequence %d", steps, resp.Sequence)
 		step.Done()
 
-		// Receieve job ids from running pipeline, use job client to attach to job stream
+		// Receive job ids from running pipeline, use job client to attach to job stream
 		// and stream here. First pass can be linear job streaming
+		successful := steps
 		for _, jobId := range resp.AllJobIds {
 			app.UI.Output("Executing Step %q", resp.JobMap[jobId].Step, terminal.WithHeaderStyle())
 			app.UI.Output("Reading job stream (jobId: %s)...", jobId, terminal.WithInfoStyle())
 			app.UI.Output("")
 
-			// Throw away the job result for now. We could do something fancy with this
-			// later.
-			_, err := jobstream.Stream(c.Ctx, jobId,
+			result, err := jobstream.Stream(c.Ctx, jobId,
 				jobstream.WithClient(c.project.Client()),
 				jobstream.WithUI(app.UI))
 			if err != nil {
 				return err
 			}
+
+			var state pb.Status_State
+			if result.Build != nil {
+				state = result.Build.Build.Status.State
+			} else if result.Deploy != nil {
+				state = result.Deploy.Deployment.Status.State
+			} else if result.Release != nil {
+				state = result.Release.Release.Status.State
+			}
+			if state != pb.Status_SUCCESS {
+				successful--
+			}
```

Comment:
I'm not sure what happened, but I got output saying 2/3 of my steps were successful but all of the jobs were successfully completed. It could be because one of them was an "exec" step and that wasn't handled right?

Reply:
I used this pipeline to test it: https://github.com/briancain/waypoint-tetris/blob/main/waypoint.hcl#L3

Reply:
Hmmm but exec doesn't have a state on it. 😵 
I'm just going to refactor this to make another API call for the job and directly check the job status, instead of trying to figure out what's going with the result object and leaving the window open for further edge cases. That way is also most fool-proof.